akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -372,8 +372,13 @@ private[remote] object EndpointManager {
      */
     def markAsFailed(endpoint: ActorRef, timeOfRelease: Deadline): Unit =
       if (isWritable(endpoint)) {
-        addressToWritable += writableToAddress(endpoint) -> Gated(timeOfRelease)
-        writableToAddress -= endpoint
+        val address = writableToAddress(endpoint)
+        addressToWritable.get(address) match {
+          case Some(Quarantined(_, _)) ⇒ // don't overwrite Quarantined with Gated
+          case _ ⇒
```

Comment:
correct (i am sorry):
>If it was already quarantined or Gated, ~~ok, that is impossible~~.

some race would happen due to async feature when actor stop.